app/api/orders.py

```diff
@@ -313,6 +318,14 @@ def before_update_object(self, order, data, view_kwargs):
             if order.order_notes and data['order_notes'] not in order.order_notes.split(","):
                 data['order_notes'] = '{},{}'.format(order.order_notes, data['order_notes'])
 
+        status = data.get('status', None)
+        if status and ((order.status == 'completed' and status in ['expired', 'initializing', 'pending', 'placed']) or
+            (order.status == 'expired' and status in ['completed', 'initializing', 'placed', 'pending']) or
+            (order.status == 'placed' and status in ['completed', 'expired', 'initializing', 'pending']) or
+                (order.status == 'pending' and status in ['initializing', 'placed'])):
```

Comment:
I think that this whole check can be optimized by making it into `if (status and order.status != status)`